app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+
+    for person in people:
+        Person(person["name"], person["age"])
+
+    for person in people:
+        person_instance = Person.people[person["name"]]
+
+        if "wife" in person and person["wife"]:
+            person_instance.wife = Person.people[person["wife"]]
```

Comment:
You could avoid having to checks with `and` by using `get()`
```suggestion
        if person.get("wife"):
            person_instance.wife = Person.people[person["wife"]]
```
and reduce the need to retrieve value by key twice by using walrus operator
```suggestion
        if wife_name := person.get("wife"):
            person_instance.wife = Person.people[wife_name]
```
Apply similar change for `husband` as well